README.md

```diff
@@ -29,20 +29,111 @@ The <b>Agglayer</b> (<i>Aggregation layer</i>) provides a common language for se
 
 </div>
 
-## Getting Started
+## Table of Contents
 
-Rust implementation of the Agglayer.
+- [Overview](#overview)
+- [Crate Directory](#crate-directory)
+- [Prerequisites](#prerequisites)
+  - [Succinct Prover Network](#succinct-prover-network)
+  - [Software Requirements](#software-requirements)
+  - [Hardware Requirements](#hardware-requirements)
+- [Installation](#installation)
+- [Running the Pessimistic Proof Test Suite](#running-the-pessimistic-proof-test-suite)
+- [Development](#development)
+- [Support](#support)
+- [Resources](#resources)
+- [License](#license)
 
-Agglayer is a web service that receives ZKPs from different CDK chains and checks the soundness of them before sending the ZKP to L1 for verification.
+## Overview
 
-To find out more about Polygon, visit the [official website](https://docs.polygon.technology/cdk/).
+AggLayer is the Rust-based service designed to (1) receive zero-knowledge proofs from AggLayer-connected chains, (2) verify their validity, and (3) send them to the L1 for final settlement. AggLayer replaces the previous Golang implementation. 
+
+To find out more about Polygon, visit the [official website.](https://docs.polygon.technology/cdk/)
 
 WARNING: This is a work in progress, and as such, all APIs and configuration are subject to change. The code is still being audited, so please contact the Polygon team if you would like to use it in production.
 
+## Crate Directory
```

Comment:
Renaming this to repository structure to begin a bit more expository section if need be. I wanna keep this for now because it gives a nice overview of the components.

Can split out once we get more complete docs.

Reply:
We can keep it for now, but we may remove this whole part and move it to rust doc as it is a more suitable place